.github/workflows/release.yaml

```diff
@@ -0,0 +1,219 @@
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Load cached deps
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+      - name: Load cached build tools
+        uses: actions/cache@v2
+        with:
+          path: .build
+          key: ${{ runner.os }}-tools-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-tools-
+      - name: Build images
+        run: make images scratch-images
+      - name: Export images
+        run: docker save spire-server:latest-local spire-agent:latest-local k8s-workload-registrar:latest-local oidc-discovery-provider:latest-local spire-server-scratch:latest-local spire-agent-scratch:latest-local k8s-workload-registrar-scratch:latest-local oidc-discovery-provider-scratch:latest-local | gzip > images.tar.gz
+      - name: Archive images
+        uses: actions/upload-artifact@v2
+        with:
+          name: images
+          path: images.tar.gz
+
+  integration:
+    runs-on: ubuntu-18.04
+    needs: [cache-deps, images]
+    strategy:
+      fail-fast: false
+      matrix:
+        num_runners: [5]
+        runner_id: [1, 2, 3, 4, 5]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Load cached deps
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+      - name: Load cached build tools
+        uses: actions/cache@v2
+        with:
+          path: .build
+          key: ${{ runner.os }}-tools-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-tools-
+      - name: Download archived images
+        uses: actions/download-artifact@v2
+        with:
+          name: images
+          path: .
+      - name: Load archived images
+        run: zcat images.tar.gz | docker load
+      - name: Run integration tests
+        env:
+          NUM_RUNNERS: ${{ matrix.num_runners }}
+          THIS_RUNNER: ${{ matrix.runner_id }}
+          TERM: dumb
+          CICD_TARGET_BRANCH: ${{ github.event.pull_request.base.ref }}
+        run: ./.github/workflows/scripts/split.sh | xargs ./test/integration/test.sh
+
+  publish-artifacts:
+    runs-on: ubuntu-18.04
+    needs: [integration]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Download archived artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: binaries
+          path: ./artifacts/
+      - name: Create Release
+        env:
+          GH_REPO: ${{ github.repository }} # https://github.com/cli/cli/issues/3556
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: gh release create "${GITHUB_REF}" ./artifacts/*.tar.gz ./artifacts/*.txt --title "${GITHUB_REF#refs/tags/}"
```

Comment:
Mind linking here in PR or in code comment to docs that explain these built-in vars? I recall having a hard time understanding the behavior I should expect from them. Probably not too critical because we can always fix up the name after release if we find that it's wrong, but good for review nonetheless

Reply:
Sure, I can add a comment or something.